File: 리트코드/0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {

        List<int[]> merged = new ArrayList<>();
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[0]));

        for (int[] i : intervals) {
            if (!merged.isEmpty() && i[0] <= merged.get(merged.size() - 1)[1]) {
                merged.get(merged.size() - 1)[1] = Math.max(i[1], merged.get(merged.size() - 1)[1]);
            } else {
                merged.add(i);
            }
        }
        return merged.toArray(new int[merged.size()][]);
    }
}

File: 리트코드/0056-merge-intervals/0056-merge-intervals.kt
class Solution {
    fun merge(intervals: Array<IntArray>): Array<IntArray> {
        intervals.sortWith(Comparator{a: IntArray, b:IntArray -> a[0] - b[0]})

        val merged: MutableList<IntArray> = mutableListOf()
        
        for (i in intervals) {
            if (merged.isNotEmpty() && i[0] <= merged[merged.size - 1][1]) {
                merged[merged.size - 1][1] = max(i[1], merged[merged.size - 1][1])
            } else {
                merged.add(i)
            }
        }
        return merged.toTypedArray()
    }
}